dlt/destinations/impl/synapse/synapse.py

```diff
@@ -0,0 +1,297 @@
+                f" {staging_table_name};"
+            )
+            # recreate staging table
+            job_client = current.pipeline().destination_client()  # type: ignore[operator]
+            with job_client.with_staging_dataset():
+                # get table columns from schema
+                columns = [c for c in job_client.schema.get_table_columns(table["name"]).values()]
+                # generate CREATE TABLE statement
+                create_table_stmt = job_client._get_table_update_sql(
+                    table["name"], columns, generate_alter=False
+                )
+            sql.extend(create_table_stmt)
+
+        return sql
+
+
+class SynapseCopyFileLoadJob(CopyRemoteFileLoadJob):
+    def __init__(
+        self,
+        table: TTableSchema,
+        file_path: str,
+        sql_client: SqlClientBase[Any],
+        staging_credentials: Optional[AzureCredentialsWithoutDefaults] = None,
+        staging_use_msi: bool = False,
+    ) -> None:
+        self.staging_use_msi = staging_use_msi
+        super().__init__(table, file_path, sql_client, staging_credentials)
+
+    def execute(self, table: TTableSchema, bucket_path: str) -> None:
+        # get format
+        ext = os.path.splitext(bucket_path)[1][1:]
+        if ext == "parquet":
+            if table_schema_has_type(table, "time"):
+                # Synapse interprets Parquet TIME columns as bigint, resulting in
+                # an incompatibility error.
+                raise LoadJobTerminalException(
+                    self.file_name(),
+                    "Synapse cannot load TIME columns from Parquet files. Switch to direct INSERT"
+                    " file format or convert `datetime.time` objects in your data to `str` or"
+                    " `datetime.datetime`",
+                )
+            file_type = "PARQUET"
+
+            # dlt-generated DDL statements will still create the table, but
+            # enabling AUTO_CREATE_TABLE prevents a MalformedInputException.
+            auto_create_table = "ON"
+        else:
+            raise ValueError(f"Unsupported file type {ext} for Synapse.")
+
+        staging_credentials = self._staging_credentials
+        assert staging_credentials is not None
+        assert isinstance(staging_credentials, AzureCredentialsWithoutDefaults)
+        azure_storage_account_name = staging_credentials.azure_storage_account_name
+        https_path = self._get_https_path(bucket_path, azure_storage_account_name)
+        table_name = table["name"]
+
+        if self.staging_use_msi:
+            credential = "IDENTITY = 'Managed Identity'"
+        else:
+            sas_token = staging_credentials.azure_storage_sas_token
+            credential = f"IDENTITY = 'Shared Access Signature', SECRET = '{sas_token}'"
+
+        # Copy data from staging file into Synapse table.
+        with self._sql_client.begin_transaction():
+            dataset_name = self._sql_client.dataset_name
+            sql = dedent(f"""
```

Comment:
out of curiosity, is dedent needed here?

Reply:
No. Functionally it doesn't make a difference, but I like the idea the query is properly formatted without unnecessary indentation.